java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java

```diff
@@ -4342,37 +4342,78 @@ void testMakeStruct() {
 
   @Test
   void testMakeListEmpty() {
-    final int numRows = 10;
-    try (ColumnVector expected =
+
+    final int numRows = 4;
+    List<List<Integer>> mainList1 = new ArrayList<>();
+    mainList1.add(Arrays.asList());
+    List<List<Integer>> mainList2 = new ArrayList<>();
+    mainList2.add(Arrays.asList());
+    List<List<Integer>> mainList3 = new ArrayList<>();
+    mainList3.add(Arrays.asList());
+    List<List<Integer>> mainList4 = new ArrayList<>();
+    mainList4.add(Arrays.asList());
+
+    try (
+        ColumnVector expectedList =
              ColumnVector.fromLists(
                  new ListType(false, new BasicType(false, DType.STRING)),
                  Arrays.asList(),
                  Arrays.asList(),
                  Arrays.asList(),
-                 Arrays.asList(),
-                 Arrays.asList(),
-                 Arrays.asList(),
-                 Arrays.asList(),
-                 Arrays.asList(),
-                 Arrays.asList(),
                  Arrays.asList());
-         ColumnVector created = ColumnVector.makeList(numRows, DType.STRING)) {
-      assertColumnsAreEqual(expected, created);
+         ColumnVector expectedListOfList = ColumnVector.fromLists(new HostColumnVector.ListType(false,
+                 new HostColumnVector.ListType(false,
+                     new HostColumnVector.BasicType(false, DType.STRING))),
+             mainList1, mainList2, mainList3, mainList4);
+
+         ColumnVector createdList = ColumnVector.makeList(numRows, DType.STRING);
+         ColumnVector createdListOfList = ColumnVector.makeList(createdList)) {
+      assertColumnsAreEqual(expectedList, createdList);
+      assertColumnsAreEqual(expectedListOfList, createdListOfList);
     }
   }
 
   @Test
   void testMakeList() {
-    try (ColumnVector expected =
-             ColumnVector.fromLists(
-                 new ListType(false, new BasicType(false, DType.INT32)),
-                 Arrays.asList(1, 3, 5),
-                 Arrays.asList(2, 4, 6));
+    List<Integer> list1 = Arrays.asList(1, 3);
+    List<Integer> list2 = Arrays.asList(2, 4);
+    List<Integer> list3 = Arrays.asList(5, 7, 9);
+    List<Integer> list4 = Arrays.asList(6, 8, 10);
+
+    List<List<Integer>> mainList1 = new ArrayList<>();
+    mainList1.add(list1);
+    mainList1.add(list3);
+    List<List<Integer>> mainList2 = new ArrayList<>();
+    mainList2.add(list2);
+    mainList2.add(list4);
```

Comment:
Nit: This could be condensed to something like:
```suggestion
    List<List<Integer>> mainList1 = new ArrayList<>(Arrays.asList(list1, list3));
    List<List<Integer>> mainList2 = new ArrayList<>(Arrays.asList(list2, list4));
```

Reply:
Done